controllers/component_build_controller_common.go

```diff
@@ -146,7 +141,6 @@ func (r *ComponentBuildReconciler) GetBuildPipelineFromComponentAnnotation(ctx c
 		}
 	}
 
-	// for now we will return PipelineRef format, because it is the same what methods which use build-selector are returning
 	pipelineRef := &tektonapi.PipelineRef{
```

Comment:
Should we stop creating pipelineRefs then?

That might also solve the suspicious-looking error handling https://github.com/konflux-ci/build-service/pull/305#discussion_r1619332287

Reply:
no because it is actually used in simple build